bundles/mapping/mapmodule/request/SetTimeRequest.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * @class SetTimeRequest
+ * SetTime Request for Cesium Map
+ */
+Oskari.clazz.define('Oskari.mapframework.request.common.SetTimeRequest',
+    /**
+     * Creates a new SetTimeRequest
+     * @param {*} date date string
+     * @param {*} time time string
+     */
+    function (date, time, year) {
+        this._date = date;
+        this._time = time;
+        this._year = year;
+    }, {
+        __name: 'SetTimeRequest',
+
+        /**
+         * @method getName
+         * @return {String} request name
+         */
+        getName: function () {
+            return this.__name;
+        },
+
+        /**
+         * @method getDate
+         * Valid format d/m
+         * @return {String} date
+         */
+        getDate: function () {
+            return this._date;
+        },
+
+        /**
+         * @method getTime
+         * @return {String} time
+         */
+        getTime: function () {
+            return this._time;
+        },
+
+        /**
+         * @method getYear
+         * @return {Number} year
+         */
+        getYear: function () {
+            return this._year || new Date().getFullYear();
+        },
+
+        /**
+         * @method validateTime
+         * @return {Bool} true if valid time
+         */
+        validateTime: function () {
+            const regex = /^(0[0-9]|1[0-9]|2[0-3]|[0-9]):[0-5][0-9]$/;
+            return regex.test(this.getTime());
+        },
+
+        /**
+         * @method validateDate
+         * @return {Bool} true if valid date
+         */
+        validateDate: function () {
+            const matches = /^(\d{1,2})[/](\d{1,2})$/.exec(this._date);
+            if (matches === null) {
+                return false;
+            }
+            const d = parseInt(matches[1]);
+            const m = matches[2] - 1;
+            const y = this.getYear();
+            const date = new Date(y, m, d);
+            return date.getDate() === d && date.getMonth() === m;
+        },
+
+        /**
+         * @method formatDate
+         * @return {String} Time formatted to ISO standard 'YYYY-MM-DDTHH:mm:ss.sssZ'
+         */
+        formatDate: function () {
+            if (!this.validateTime() || !this.validateDate()) {
+                return false;
+            }
```

Comment:
Should these be checked in handler instead?

Reply:
Yes, move the validation to handler.